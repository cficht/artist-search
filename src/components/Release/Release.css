.Release {
  display: flex;
  justify-content: center;  
  align-items: center;
  flex-direction: column;

  background-color: red;
  color: white;

  text-align: center;
}

.List {
  display: flex;
  justify-content: center;

  width: 30%;
  margin: 10px 20px 20px 20px;
  padding: 20px;
  border-radius: 10px;

  background-color: darkblue;

  ul {
    margin-block-start: 0em;
    margin-block-end: 0em;
    padding-inline-start: 0px;
    list-style-type: none;
  }

  li {
    margin: 10px;
  }

  a {
    color: white;
    font-size: 16px;
  }
}