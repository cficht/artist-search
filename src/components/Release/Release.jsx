import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Release = ({ artistName, recordings }) => {
  const recordingNodes = recordings.map(recording => (
    <li key={recording.id}>
      <Link to={`/recording/${artistName}/${recording.title}`}>{recording.title}</Link>
    </li>
  ));

  return (
    <section>
      <h2>{artistName}</h2>
      <ul>
        {recordingNodes}
      </ul>
    </section>
  );
};

Release.propTypes = {
  artistName: PropTypes.string.isRequired,
  recordings: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
  })).isRequired
};

export default Release;
