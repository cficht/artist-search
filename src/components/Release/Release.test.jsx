import React from 'react';
import { shallow } from 'enzyme';
import Release from './Release';

describe('Release component', () => {
  it('renders Release', () => {
    const wrapper = shallow(<Release 
      artistName="The Beatles"
      recordings={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
