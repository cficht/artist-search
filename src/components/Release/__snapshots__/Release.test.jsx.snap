// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Release component renders Release 1`] = `
<section
  className="Release"
>
  <h2>
    The Beatles
  </h2>
  <article
    className="List"
  >
    <ul />
  </article>
</section>
`;
